api/app/schemas/shared.py

```diff
@@ -0,0 +1,11 @@
+""" This module contains shared pydantic schemas.
+"""
+
+from typing import List
+from pydantic import BaseModel
+
+
+class WeatherDataRequest(BaseModel):
+    """ A request for weather data for a given set of stations with a time of interest. """
+    stations: List[int]
+    time_of_interest: str = None
```

Comment:
How about defaulting to now as the time of interest? Then you don't need all that code checking for null and inserting it.
```suggestion
    time_of_interest: str = get_utc_now().isoformat()
```

Reply:
So the problem with this is that mocking `get_utc_now` doesn't work at all here. I couldn't figure out why so I went the non-elegant way. Thought?